crates/types/src/traits/block_contents.rs

```diff
@@ -47,7 +53,15 @@ pub trait BlockPayload:
     type Transaction: Transaction;
 
```

Comment:
Should we remove the `encode` method and add a `EncodeBytes` supertrait instead?

Reply:
I think we should, but `encode` method is fallible and I think `EncodeBytes` shouldn't be.
I edited the PR description with explanation, pasting it here:
> ### This PR does not: 
> Replace `BlockPayload::encode` with bound on `EncodeBytes`, since `BlockPayload::encode` is fallible.
In my opinion, `encode` shouldn't be fallible and we need to push validation to the point of structure creation, because non-encodable `BlockPayload` is essentially useless and shouldn't be representable.

Reply:
Gotcha. `Payload::encode` in the sequencer is actually infallible, so it's probably fine to change this. But we can do it in a separate change

Reply:
Added #3020